p2p/protocol/identify/id_glass_test.go

```diff
@@ -173,3 +174,34 @@ func TestInvalidSignedPeerRecord(t *testing.T) {
 	require.True(t, ok)
 	require.Nil(t, cab.GetPeerRecord(h2.ID()))
 }
+
+func TestIncomingAddrFilter(t *testing.T) {
+	lhAddr := ma.StringCast("/ip4/127.0.0.1/udp/123/quic-v1")
+	privAddr := ma.StringCast("/ip4/192.168.1.101/tcp/123")
+	pubAddr := ma.StringCast("/ip6/2::1/udp/123/quic-v1")
+	dnsAddr := ma.StringCast("/dns/example.com/udp/123/quic-v1")
```

Comment:
Should we add a `/dns/localhost` address here?

Reply:
I'd prefer adding this when we start using the updated functions in [go-multiaddr](https://github.com/multiformats/go-multiaddr/pull/221) package.

Reply:
Here's the v0.12 release PR: https://github.com/multiformats/go-multiaddr/pull/223

Reply:
v0.12 was released: https://github.com/multiformats/go-multiaddr/releases/tag/v0.12.0

@sukunrt Can you update this PR?